### PR TITLE
feat: Add `SpeculativeExecutionScheduling` new algorithm with tests

### PR DESCRIPTION
**Pull request by Hardvan**
- [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Java/blob/master/CONTRIBUTING.md).
- [x] This pull request is all my own work -- I have not plagiarized it.
- [x] All filenames are in PascalCase.
- [x] All functions and variable names follow Java naming conventions.
- [x] All new algorithms have a URL in their comments that points to Wikipedia or other similar explanations.
- [x] All new code is formatted with `clang-format -i --style=file path/to/your/file.java`

### CODE REVIEW

File: src/main/java/com/thealgorithms/scheduling/SpeculativeExecutionScheduling.java [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+/**
+ * SpeculativeExecutionScheduling runs multiple copies of the same task in parallel,
+ * picking the first one that finishes successfully. It is used to mitigate the
+ * impact of stragglers (slow tasks).
+ *
+ * Use Case: Big data systems like Hadoop and Spark, where it helps improve job completion time.
+ *
+ * @author Hardvan
+ */
+public final class SpeculativeExecutionScheduling {
```
Comment (siriak) [inline src/main/java/com/thealgorithms/scheduling/SpeculativeExecutionScheduling.java:17]: Scheduling should assign start time to all tasks, but this algorithm doesn't do that

**Review by siriak — CHANGES_REQUESTED** [review]

File: src/main/java/com/thealgorithms/scheduling/SpeculativeExecutionScheduling.java (hunk 1 above)
Comment (Hardvan) [reply src/main/java/com/thealgorithms/scheduling/SpeculativeExecutionScheduling.java:17]: Done! I added the `startTime` variable.

File: src/main/java/com/thealgorithms/scheduling/SpeculativeExecutionScheduling.java [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+
+    static class Task {
+        String name;
+        boolean completed;
+        long startTime;
+
+        Task(String name) {
+            this.name = name;
+            this.completed = false;
+            this.startTime = -1;
```
Comment (siriak) [inline src/main/java/com/thealgorithms/scheduling/SpeculativeExecutionScheduling.java:27]: It should be the time when the task arrives to the system and becomes available for execution.

File: src/main/java/com/thealgorithms/scheduling/SpeculativeExecutionScheduling.java [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+        tasks.add(new Task(taskName));
+    }
+
+    /**
+     * Executes the tasks in the specified group by assigning a start time.
+     *
+     * @param groupName the name of the group
+     * @return the name of the task that completed successfully
+     */
+    public String executeTasks(String groupName) {
```
Comment (siriak) [inline src/main/java/com/thealgorithms/scheduling/SpeculativeExecutionScheduling.java:66]: Please provide reference explanation of this algorithm

**Review by siriak — CHANGES_REQUESTED** [review]

**Pull request closed without merging**